### PR TITLE
Large mint borrow anomaly detection

### PR DESCRIPTION
**Pull request by notyouraveragedev**
Finished bot and it keeps track of mint/borrow amount per account

### CODE REVIEW

File: large-mint-borrow-anomaly-detection/src/agent.js [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+    bucketBlockSize,
+    ARIMA_CONFIG
+  );
+
+  trackerBuckets.push(timeSeriesTemp);
+};
+
+const alreadyTracked = (address, trackerBuckets) => {
+  const trackerKeys = [];
+  for (let bucket of trackerBuckets) {
```
Comment (dikel) [inline large-mint-borrow-anomaly-detection/src/agent.js]: Why do we need to push the addresses first and then find the index? Can't we iterate with index and if the address is equal to trackerBuckets[i] => return the index?

File: large-mint-borrow-anomaly-detection/src/agent.js [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+const {
+  commonEventSigs,
+  limitTracked,
+  getMinBucketBlockSizeByChainId,
+  getBlocktimeByChainId,
+} = require("./agent.config");
+
+let { bucketBlockSize } = require("./agent.config");
+
+const ARIMA_CONFIG = {
```
Comment (dikel) [inline large-mint-borrow-anomaly-detection/src/agent.js:18]: Have you tested the model with real data? Are these the best params?

File: large-mint-borrow-anomaly-detection/src/agent.js [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+  verbose: false,
+};
+const ADDRESS_ZERO = ethers.constants.AddressZero;
+const TimeAnomalyDetection = require("./TimeAnomalyDetection");
+const trackerBuckets = [];
+const provider = getEthersProvider();
+
+let isRunningJob = false;
+let localFindings = [];
+let aggregationTimePeriod = 1; //The period that is calculated by bucketBlockSize * blockTime
```
Comment (dikel) [inline large-mint-borrow-anomaly-detection/src/agent.js]: This should be in a bot-config file

File: large-mint-borrow-anomaly-detection/src/agent.js [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+            name: "Large mint volume",
+            description: `${findingData.from_address} minted an unusually high number of ${findingData.numberOfAssets} assets ${findingData.mintedAssetAccount}`,
+            alertId: "HIGH-MINT-VALUE",
+            severity: FindingSeverity.Medium,
+            type: FindingType.Exploit,
+            metadata: {
+              FIRST_TRANSACTION_HASH: findingData.firstTxHash,
+              LAST_TRANSACTION_HASH: findingData.lastTxHash,
+              ASSET_IMPACTED: findingData.mintedAssetAccount,
+              BASELINE_VOLUME: Math.floor(low),
```
Comment (dikel) [inline large-mint-borrow-anomaly-detection/src/agent.js]: Shouldn't the baseline be the predicted value?

File: large-mint-borrow-anomaly-detection/src/agent.js [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+            name: "Large borrow volume",
+            description: `${findingData.from_address} borrowed an unusually high number of ${findingData.numberOfAssets} assets ${findingData.borrowedAssetAccount}`,
+            alertId: "HIGH-BORROW-VALUE",
+            severity: FindingSeverity.Medium,
+            type: FindingType.Exploit,
+            metadata: {
+              FIRST_TRANSACTION_HASH: findingData.firstTxHash,
+              LAST_TRANSACTION_HASH: findingData.lastTxHash,
+              ASSET_IMPACTED: findingData.borrowedAssetAccount,
+              BASELINE_VOLUME: low,
```
Comment (dikel) [inline large-mint-borrow-anomaly-detection/src/agent.js]: Shouldn't the baseline be the predicted value?

**Review by dikel — CHANGES_REQUESTED** [review]

**Review by dikel — APPROVED** [review]
LGTM

File: large-mint-borrow-anomaly-detection/README.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  - Type is always set to "exploit"
+  - Metadata fields:
+    - FIRST_TRANSACTION_HASH (first hash when it occured)
+    - LAST_TRANSACTION_HASH (last hash when it occured)
+    - ASSET_IMPACTED (address of the impaced asset)
+    - BASELINE_VOLUME (the normal volume)
+
+## Test Data
+
+The bot behaviour can be verified with supplied unit tests
```
Comment (christian-forta) [inline large-mint-borrow-anomaly-detection/README.md:43]: do we have a tx or block range for a real world attack we can reference here?

**Review by christian-forta — APPROVED** [review]

**Pull request merged**